ProQuest.js

```diff
@@ -227,34 +249,24 @@ function doWeb(doc, url, pdfUrl) {
 				ZU.processDocuments(articles, doWeb);
 			}
 		});
-	//pdfUrl should be undefined unless we are calling doWeb from the following
-	//block, where it is set to false or an actual value
-	} else if(followLink && pdfUrl === undefined) {
-		pdfUrl = false;
-		var link = ZU.xpathText(doc,
-			'//a[@class="formats_base_sprite format_abstract"]/@href');
-		if(!link) return;
-
-		//see if we can get the full text PDF link before we go
-		//the logic here is actually slightly different from fetchEmbeddedPdf
-		if (pdfUrl = doc.getElementById('EmbedFile')
-				|| doc.getElementsByTagName('embed')[0]
-		) {
-			pdfUrl = pdfUrl.src;
-		} else {
-			pdfUrl = false;
+	} else if (!followLink) {//i.e. we are on an abstract/metadata page
+		scrape(doc, url, type);
+	} else {//i.e. we are on pdf page
+		var link = ZU.xpathText(doc, '//li[@id="tab-AbstractRecord-null"]/a/@href');
+		if (!link) {
+			Z.debug("Error: Could not find the abstract/metadata page");
+			return;
 		}
-
 		ZU.processDocuments(link, function(doc) {
-			doWeb(doc, doc.location.href, pdfUrl) });
+			doWeb(doc, doc.location.href) });
```

Comment:
`ZU.processDocuments(link, function(doc, url) { doWeb(doc, url, true) })` and add a third parameter to `doWeb` that's something like `noFollow` to make sure we don't get stuck in an infinite loop here.